### PR TITLE
Room booking

### PR DESCRIPTION
**Pull request by lannp**
https://dev.framgia.com/redmine/issues/60864

### CODE REVIEW

File: app/scripts/controllers/room_booking_controller.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('app.controllers')
+    .controller('RoomBookingController', RoomBookingController);
+
+  RoomBookingController.$inject = ['DataFactory', '$firebaseArray', '$window'];
```
Comment (xuanduc987) [inline app/scripts/controllers/room_booking_controller.js]: Seems like missing `EventsService`

File: app/scripts/app.js [hunk 2]
```diff
@@ -77,6 +77,11 @@
         controller: 'RoomsController',
         controllerAs: 'room'
       })
+      .when('/rooms/book', {
+        templateUrl: 'views/rooms/book.html',
+        controller: "RoomBookingController",
```
Comment (yeuem1vannam) [inline app/scripts/app.js]: This should be defined in `EventsController` -> action `createEvent`

File: app/scripts/app.js [hunk 3]
```diff
@@ -82,6 +82,12 @@
         controller: 'ProjectsController',
         controllerAs: 'project'
       })
+      .when('/events/new', {
+        templateUrl: 'views/events/new.html',
+        controller: 'EventsController',
+        controllerAs: 'vm'
```
Comment (yeuem1vannam) [inline app/scripts/app.js]: This name should be related to controller name. It is not related to `vm` variable which defined inside controller.

File: app/scripts/controllers/events_controller.js [hunk 4]
```diff
@@ -41,5 +55,45 @@
...
+          projectName: vm.booking.projectName,
+          description: vm.booking.description,
+          timeFrom: timeFrom,
+          timeTo: timeTo,
+          userName: "abc"
+        };
+
+        EventsService.create(booking).then(
+        function() {
+          $window.location.href = '#/events';
```
Comment (xuanduc987) [inline app/scripts/controllers/events_controller.js:75]: Dùng `$location` đi

**Comment by yeuem1vannam** [comment]
Merge cho các task sau chạy.
Move `$location` sang https://dev.framgia.com/redmine/issues/60866

**Pull request merged**